components/infobox/wikis/warthunder/infobox_league_custom.lua

```diff
@@ -0,0 +1,67 @@
+---
+-- @Liquipedia
+-- wiki=warthunder
+-- page=Module:Infobox/League/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Injector = Lua.import('Module:Widget/Injector')
+local League = Lua.import('Module:Infobox/League')
+
+local Widgets = require('Module:Widget/All')
+local Cell = Widgets.Cell
+
+---@class WarThunderLeagueInfobox: InfoboxLeague
+local CustomLeague = Class.new(League)
+local CustomInjector = Class.new(Injector)
+
+-- Battle Mode: Different Battle Mode each providing different pacing.
+local BATTLEMODES = {
+	ab = 'Arcade',
+	rb = 'Realistic',
+	sb = 'Simulator',
+	rbm = 'Realistic with Markers',
+}
+-- Vehicle: Type of units played in the event.
+local VEHICLES = {
+	aviation = 'Aviation',
+	ground = 'Ground',
+	naval = 'Naval',
+	multiple = 'Multiple',
+}
+
+---@param frame Frame
+---@return Html
+function CustomLeague.run(frame)
+	local league = CustomLeague(frame)
+	league:setWidgetInjector(CustomInjector(league))
+
+	return league:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	local caller = self.caller
+	local args = caller.args
+
+	if id == 'custom' then
+		local battleMode = BATTLEMODES[string.lower(args.battlemode or '')] or 'Unknown'
```

Comment:
Why use "Battle Mode" over "Mode"?

Reply:
> Why use "Battle Mode" over "Mode"?

idk if it would conflict the stored team/solo input here 
![Uploading image.png…]()

Reply:
![image](https://github.com/user-attachments/assets/457dedb6-dd65-48d1-aace-e06d79f78564)

Reply:
alright then the current revision should now work as intended then, its using modes now